File: Backend-zoo-app/models/Animal.js
const mongoose = require('mongoose');

const animalSchema = new mongoose.Schema({
    nombre: { type: String, required: true },
    descripcion: { type: String }, // ej: Biología y Comportamiento
    gradoAmenaza: { type: String },
    dieta: { type: String },
    curiosidades: { type: String },
    vida: { type: String },
    ecosistema: { type: String }, // ej: 'África ecuatorial'
    tipoAnimal: { type: String }, // ej: mamífero
    imagen: { type: String }, //URL de ImageKit
    zonaId: { type: mongoose.Schema.Types.ObjectId, ref: 'Zona', required: true }
});

module.exports = mongoose.model('Animal', animalSchema);

File: Frontend-ZooApp/app/src/main/java/com/example/zooapp/Animal.kt
package com.example.zooapp

data class Animal(
    val _id: String? = null,
    val nombre: String,
    val descripcion: String,
    val gradoAmenaza: String = "",
    val dieta: String = "",
    val curiosidades: String = "",
    val vida: String = "",
    val ecosistema: String,
    val tipoAnimal: String,
    val interesTematico: List<String> = emptyList(),
    val imagen: String? = null,
    val zonaId: String
)